momento-sdk/src/main/java/momento/sdk/ScsTopicGrpcStubsManager.java

```diff
@@ -100,8 +135,30 @@ PubsubGrpc.PubsubStub getNextUnaryStub() {
   }
 
   /** Round-robin subscribe stub. */
-  PubsubGrpc.PubsubStub getNextStreamStub() {
-    return streamStubs.get(streamIndex.getAndIncrement() % this.numStreamGrpcChannels);
+  StreamStubWithCount getNextStreamStub() {
+    // First check if there's available capacity on any of the stubs
+    int totalActiveSubscriptions = 0;
+    for (StreamStubWithCount stubWithCount : streamStubs) {
+      totalActiveSubscriptions += stubWithCount.getCount();
+    }
+    if (totalActiveSubscriptions < this.maximumActiveSubscriptions) {
+      // Try to get a client with capacity for another subscription.
+      // Allow up to maximumActiveSubscriptions attempts.
+      for (int i = 0; i < this.maximumActiveSubscriptions; i++) {
+        // Round-robin to the next stub
+        final StreamStubWithCount stubWithCount =
+            streamStubs.get(streamIndex.getAndIncrement() % this.numStreamGrpcChannels);
+        if (stubWithCount.getCount() < 100) {
+          stubWithCount.incrementCount();
```

Comment:
There's a TOCTOU bug here. The stubWithCount should have an acquire or fail method that atomically acquires a resource (tallies one to the count) ro else fails. When we fail we try a different channel.